src/types/events/base-events.ts

```diff
@@ -76,16 +76,13 @@ export type SlackEvent =
  * this interface. That condition isn't enforced, since we're not interested in factoring out common properties from the
  * known event types.
  */
-export interface BasicSlackEvent<Type extends string = string> extends StringIndexed {
+export interface BasicSlackEvent<Type extends string = string> {
   type: Type;
 }
 
 /* ------- TODO: Generate these interfaces ------- */
 
-// TODO: why are these all StringIndexed? who does that really help when going more than one level deep means you have
```

Comment:
Once I removed `StringIndexed` as a base from all these interfaces, many unsafe property accesses in our own codebase became type-checking errors. Most of refactoring in the rest of this PR is simply an attempt to address those errors.